Test/Unit/Helper/Webhook/AuthorisationWebhookHandlerTest.php

```diff
@@ -1,130 +1,363 @@
+            $paymentMethodsMock
+        );
+
+        // Invoke the private method
+        $handleSuccessfulAuthorisationMethod = $this->getPrivateMethod(
+            AuthorisationWebhookHandler::class,
+            'handleSuccessfulAuthorisation'
         );
 
-        $transitionState = PaymentStates::STATE_FAILED;
-        $result = $handler->handleWebhook($order, $notification, $transitionState);
+        $result = $handleSuccessfulAuthorisationMethod->invokeArgs($authorisationWebhookHandler, [$this->orderMock, $this->notificationMock]);
+
+        // Assert the result
+        $this->assertInstanceOf(Order::class, $result);
+    }
 
-        // Assert that handleFailedAuthorisation was called and handleSuccessfulAuthorisation wasn't called
-        $this->assertNotNull($result);
-        $this->assertEquals($order->toArray(), $result->toArray());
+    public function testHandleFailedAuthorisation()
+    {
+        $this->orderMock->expects($this->atLeastOnce())
+            ->method('getData')
+            ->withConsecutive(
+                ['adyen_notification_event_code'],
+                ['adyen_notification_payment_captured']
+            )
+            ->willReturnCallback(function ($arg) {
+                if ($arg === 'adyen_notification_event_code') {
+                    return 'AUTHORISATION : TRUE';
+                } elseif ($arg === 'adyen_notification_payment_captured') {
+                    return false;
+                }
+            });
 
+        // Create an instance of AuthorisationWebhookHandler
+        $webhookHandler = $this->createAuthorisationWebhookHandler(
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null
+        );
+
+        $handleFailedAuthorisationMethod = $this->getPrivateMethod(
+            AuthorisationWebhookHandler::class,
+            'handleFailedAuthorisation'
+        );
+
+        // Call the private method directly and provide required parameters
+        $result = $handleFailedAuthorisationMethod->invokeArgs($webhookHandler, [$this->orderMock, $this->notificationMock]);
+
+        // Assert the expected behavior based on the mocked logic and result
+        $this->assertInstanceOf(Order::class, $result);
+    }
+
+    public function testHandleAutoCapture()
+    {
+        // Set up expectations for the mocks
+        $this->orderMock->expects($this->any())
+            ->method('getPayment')
+            ->willReturn($this->orderMock);
+
+        $this->orderMock->expects($this->any())
+            ->method('getConfig')
+            ->willReturnSelf();
+
+        // Create an instance of AuthorisationWebhookHandler
+        $webhookHandler = $this->createAuthorisationWebhookHandler(
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null
```

Comment:
You can skip sending values for optional arguments. They are already `null` by default.